pkg/ibmvpc/nacl_analysis.go

```diff
@@ -445,6 +461,78 @@ func (na *NACLAnalyzer) AllowedConnectivity(subnetCidr, inSubentCidr, target str
 		}
 	}
 	// expecting disjoint ip-blocks, thus not expecting to get here
-	return nil, fmt.Errorf("isIngress: %t , target %s, subnetCidr: %s, inSubentCidr %s,  "+
-		"could not find allowed connectivity for given target + inSubentCidr", isIngress, target, subnetCidr, inSubentCidr)
+	return nil, fmt.Errorf(notFoundMsg, isIngress, target, subnetCidr, inSubentCidr)
+}
+
+// rulesInConnectivity returns set of rules contributing to a connections given src/dst and direction
+// if conn is specified then rules contributing to that connection; otherwise to any connection src->dst
+// if the input subnet was not yet analyzed, it first adds its analysis to saved results
+func (na *NACLAnalyzer) rulesInConnectivity(subnetCidr, inSubentCidr,
+	target string, connQuery *common.ConnectionSet, isIngress bool) ([]int, error) {
+	// add analysis of the given subnet
+	// analyzes per subnet disjoint cidrs (it is not necessarily entire subnet cidr)
+	analyzedConns, targetIPblock, inSubnetIPblock := na.initConnectivityRelatedCompute(subnetCidr, inSubentCidr, target, isIngress)
+
+	for disjointSubnetCidr, analyzedConnsPerCidr := range analyzedConns {
+		disjointSubnetCidrIPblock, err := common.IPBlockFromIPRangeStr(disjointSubnetCidr)
+		if err != nil {
+			return nil, err
+		}
+		if inSubnetIPblock.ContainedIn(disjointSubnetCidrIPblock) {
+			for resTarget, rules := range analyzedConnsPerCidr.contribRules {
+				if !targetIPblock.ContainedIn(resTarget) {
+					continue
+				}
+				if connQuery == nil {
+					return rules, nil
+				}
+				// connection is part of the query
+				contained, err := connQuery.ContainedIn(analyzedConnsPerCidr.allowedconns[resTarget])
+				if err != nil {
+					return nil, err
+				}
+				if contained {
+					return na.getRulesRelevantConn(rules, connQuery) // gets only rules relevant to conn
+				}
```

Comment:
if the connection is part of the query, but it contains multiple ports, and it only intersects certain ports of a certain rule, then this rule is not considered as contributing to the connection?